librepo/handle.c

```diff
@@ -264,6 +266,32 @@ lr_handle_remote_sources_changed(LrHandle *handle, LrChangedRemoteSource type)
     }
 }
 
+struct callback_data {
+   GMainLoop *loop; 
+   long input_seconds;
+   time_t begin;
```

Comment:
How about just storing e.g.
```suggestion
    guint64 deadline;
```

which we compute once via:

```
  data_struct.deadline_secs = g_get_monotonic_time() + seconds * G_USEC_PER_SEC;
```